test/unit/selector.js

```diff
@@ -2142,3 +2142,59 @@ QUnit.test( "jQuery.escapeSelector", function( assert ) {
 	assert.equal( jQuery.escapeSelector( "\uDF06" ), "\uDF06", "Doesn't escape lone low surrogate" );
 	assert.equal( jQuery.escapeSelector( "\uD834" ), "\uD834", "Doesn't escape lone high surrogate" );
 } );
+
+QUnit.test( "custom pseudos", function( assert ) {
+	assert.expect( 6 );
+
+	jQuery.expr.filters.foundation = jQuery.expr.filters.root;
+	assert.deepEqual( jQuery.find( ":foundation" ), [ document.documentElement ], "Copy element filter with new name" );
+	delete jQuery.expr.filters.foundation;
```

Comment:
For good hygiene, I think we should wrap this in a `try` block move the cleanup into `finally` (or anything else that accomplishes the same isolation).

Reply:
@gibson042 I used `try-finally` for each part. Does that look OK now?

Reply:
👍